agent/client.go

```diff
@@ -92,13 +92,18 @@ func newClient(addr, serverName, caFile, certFile, keyFile string) (*client, err
 		},
 	})
 
-	return NewClientWithCreds(addr, creds)
+	return NewClientWithCreds(ctx, addr, creds)
 }
 
 // NewClientWithCreds creates a new agent RPC client to the given address
 // using specified credentials creds
-func NewClientWithCreds(addr string, creds credentials.TransportCredentials) (*client, error) {
-	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(creds))
+func NewClientWithCreds(ctx context.Context, addr string, creds credentials.TransportCredentials) (*client, error) {
+	conn, err := grpc.DialContext(
+		ctx,
+		addr,
+		grpc.WithTransportCredentials(creds),
+		grpc.WithBlock(),
```

Comment:
I'm curious, why this change to block until connection is available? I wonder if that could cause any issues with palnet-agent startup.

Reply:
I thought it would be easier to handle dial connection problems, but I guess it's not necessary.